aspnet/mvc/models/file-uploads.rst

```diff
@@ -0,0 +1,197 @@
+  :language: c#
+  :emphasize-lines: 16-19
+
+Angular automatically passes an antiforgery token in a request header named ``X-XSRF-TOKEN``; the ASP.NET Core MVC app is configured to refer to this header in its configuration in ``Startup.cs``:
+
+.. literalinclude:: file-uploads/sample/FileUploadSample/Startup.cs
+  :language: c#
+  :lines: 10-16
+  :emphasize-lines: 3-4
+
+The ``DisableFormValueModelBinding`` attribute is used to disable model binding for the ``Upload`` action method:
+
+.. literalinclude:: file-uploads/sample/FileUploadSample/Filters/DisableFormValueModelBindingAttribute.cs
+  :language: c#
+  :emphasize-lines: 18,26
+
+Since model binding is disabled, the ``Upload`` action method doesn't accept any parameters. It works directly with the ``Request`` property of ``ControllerBase``. A ``MultipartReader`` is used to read each section, and either save the file (with a GUID filename) or accumulate the key/value data in a ``KeyValueAccumulator``. Once all sections have been read, the contents of the ``KeyValueAccumulator`` are used to bind the form data to a model type.
+
+The complete ``Upload`` method is shown below:
+
+.. literalinclude:: file-uploads/sample/FileUploadSample/Controllers/StreamingController.cs
+  :language: c#
+  :lines: 43-155
+  :dedent: 8
+  :emphasize-lines: 6-7,17,25,32,42,44,51,72,95-96,105-111
+  
+Troubleshooting
+---------------
+
+Below are some common problems encountered when working with uploading files, and their possible solutions.
+
+Unexpected Not Found Error with IIS
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The following error indicates your file upload exceeds the server's configured ``maxAllowedContentLength``:
+
+.. code-block:: text
+
+  HTTP 404.13 - Not Found
+  The request filtering module is configured to deny a request that exceeds the request content length.
+
+The default setting is ``30000000``, which is approxmiately 28.6MB. The value can be customized by editing ``web.config``:
+
+.. code-block:: xml
+
+  <system.webServer>
+    <security>
+      <requestFiltering>
+        <!-- This will handle requests up to 50MB -->
+        <requestLimits maxAllowedContentLength="52428800" />
+      </requestFiltering>
+    </security>
+  </system.webServer>
+
+This setting only applies to IIS; the behavior doesn't occur by default when hosting on Kestrel. `Learn more <https://www.iis.net/configreference/system.webserver/security/requestfiltering/requestlimits>`_.
+
+Null Reference Exception with IFormFile
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If your controller is accepting uploaded files using ``IFormFile``, but you find that the value is always null, be sure that your HTML form is specifying an ``enctype`` value of ``multipart/form-data``. If this attribute is not set on the ``<form>`` element, the file upload will not occur (and any bound ``IFormFile`` arguments will be null).
+
+.. tip:: HTML file uploads must specify ``<form enctype="multipart/form-data">``.
```

Comment:
For emphasis or those who are only skimming. A la callouts in magazine articles where a particular quote or line is emphasized. Not desired?